### PR TITLE
Remove residual code related to payment method promotions.

### PR DESCRIPTION
**Pull request by mordeth**
#### Changes proposed in this Pull Request
This pull request removes the superfluous residual code that was previously implemented with the #6905.

#### Testing instructions
This PR doesn't come with explicit testing guidelines. Nevertheless, we can validate the following:
- The payment methods list is presented without encountering any errors.
- The unit tests are passing successfully.

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [x] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `7585` or branch name `remove/jcb-promotion-leftovers` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&woocommerce-payments-dev-tools&branches.woocommerce-payments=remove/jcb-promotion-leftovers) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: a4d9cca878aac7a99fc5c2769a9a262fd8ac8c13
- Build time: 2023-11-03 18:03:34 UTC

_Note: the build is updated when a new commit is pushed to this PR._

**Review by KarlisJ — APPROVED** [review]
LGTM!

Tested the settings screen with some pending capabilities to assert that tooltips are displayed as expected.

File: changelog/remove-jcb-promotion-leftovers [hunk 1]
```diff
@@ -0,0 +1,4 @@
+Significance: minor
+Type: dev
+
+Remove unnecessary residual code related to payment method promotions.
```
Comment (anu-rock) [inline changelog/remove-jcb-promotion-leftovers]: It's strangely showing me a red minus icon below the changelog entry:

Screenshot 2023-11-01 at 3 59 19 PM

**Review by anu-rock — APPROVED** [review]
LGTM. Thanks for the trash cleanup 🙌

File: changelog/remove-jcb-promotion-leftovers (hunk 1 above)
Comment (mordeth) [reply changelog/remove-jcb-promotion-leftovers]: The issue was due to the absence of an empty newline. Resolved.

**Review by mgascam — CHANGES_REQUESTED** [review]
Thanks for addressing this @mordeth! 

I noticed this error in the Payments > Settings screen.

Screenshot 2023-11-03 at 12 55 38

I believe we need to remove the `isSetupRequired` prop from here https://github.com/Automattic/woocommerce-payments/blob/b011b37dd8c7bb16ef7c75d12ad681f2d3585135/client/components/payment-methods-checkboxes/payment-method-checkbox.tsx#L131.

Can you check it on your end please?

**Comment by mordeth** [comment]
@tpaksu recently merged a pull request #7211  that used the outdated code from the previous JCB promotion integration, which we aimed to eliminate with the current pull request. I'm closing the PR because the code is intended for future use.

**Pull request closed without merging**